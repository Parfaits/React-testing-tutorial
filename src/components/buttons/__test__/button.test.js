import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

import Button from '../button';
import CustomButton from '../customButton';

test('should render without crashing', () => {
    render(<Button />);
    render(<CustomButton text='hit me'/>);
});

test('should expect button to be active', () => {
    render(<Button />);
    const button = screen.getByRole('button');
    expect(button).not.toBeDisabled();
});

test('should fire click event', () => {
    const onClick = jest.fn(); // mock function
    render(<Button onClick={onClick}/>);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(onClick).toBeCalled();

    // wait until the `get` request promise resolves and
    // the component calls setState and re-renders.
    // `waitFor` waits until the callback doesn't throw an error
    // await waitFor(() =>
    // // getByRole throws an error if it cannot find an element
    //     screen.getByRole('heading'),
    // )
});

// Snapshot testing with button
test('should change text when mouse hovered', () => {
    // create initial snapshot
    const component = renderer.create(<CustomButton text='click meeee'/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // snapshot when mouse hovers over button; should change text to 'hovered'
    tree.props.onMouseEnter();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // snapshot when mouse leaves button; should change text back to original text
    tree.props.onMouseLeave();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });