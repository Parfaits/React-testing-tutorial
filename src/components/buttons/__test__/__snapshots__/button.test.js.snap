// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change text when mouse hovered 1`] = `
<div
  className="custom-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="button"
>
  click meeee
</div>
`;

exports[`should change text when mouse hovered 2`] = `
<div
  className="custom-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="button"
>
  hovered
</div>
`;

exports[`should change text when mouse hovered 3`] = `
<div
  className="custom-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="button"
>
  click meeee
</div>
`;
